### PR TITLE
Strip transformer prefix in addr mesh requests to namerd

### LINKED ISSUE #1660: io.l5d.mesh interpreter fails to resolve daemonset transformer
Issue Type:

- [x] Bug report
- [ ] Feature request

**What happened**:
See https://discourse.linkerd.io/t/linkerd-namerd-cni-setup-help/313 for additional details.
In a linkerd + namerd CNI setup, namerd successfully resolves the daemonset transformer, but linkerd does not.

namerd:
screen shot 2017-10-02 at 4 52 26 pm
linkerd:
screen shot 2017-10-02 at 4 52 15 pm

**Environment**:
- linkerd/namerd version, config files: v 1.2.1
- Platform, version, and config files (Kubernetes, DC/OS, etc): kubernetes + flannel CNI

### PR DESCRIPTION
**Pull request by adleong**
Fixes #1660 

When Linkerd uses the namerd mesh interpreter and namerd has namers that use transformers, the bound names that namerd returns will include transformer prefixes e.g. `/%/io.l5d.k8s.daemonset/linkerd/http/l5d/#/ds/default/http/hello`.  When this bound id is passed to the resolve endpoint, the transformer prefix is still present and therefore the id will not match any namers and the call will return Addr.Neg.  This results in a NoHostsAvailable exception.

We have the resolve endpoint strip off any transformer prefix from the bound id before attempting to match it against the namers.  This is what the thrift namer interface already does.

### CODE REVIEW

File: finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4StreamTransport.scala [hunk 1]
```diff
@@ -184,9 +184,9 @@ private[h2] trait Netty4StreamTransport[SendMsg <: Message, RecvMsg <: Message]
   private[this] val stateRef: AtomicReference[StreamState] = {
     val remoteMsgP = new Promise[RecvMsg] with Promise.InterruptHandler {
       override protected def onInterrupt(t: Throwable): Unit =
-      // When the remote message--especially a client's response--is
-      // canceled, close the transport, sending a RST_STREAM as
-      // appropriate.
+        // When the remote message--especially a client's response--is
+        // canceled, close the transport, sending a RST_STREAM as
+        // appropriate.
```
Comment (hawkw) [inline finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4StreamTransport.scala]: This looks like an unrelated change (although the new indentation is more correct... ¯\_(ツ)_/¯)

File: k8s/src/main/scala/io/buoyant/k8s/resources.scala [hunk 2]
```diff
@@ -91,18 +91,18 @@ private[k8s] class NsVersion[O <: KubeObject](
...
+ * various naming standards have been changed to make the apis more meaningful. For more details
+ * see https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-third-party-resource/.
+ * To understand how to create your own custom resources see
+ * https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/ for a description of the
+ * design model.
+ *
+ * Implementors are likely to want to extend this to provide convenience methods on top of the `list` method.
+ *
+ * @tparam O The parent type for all [[KubeObject]]s served by this Version.
+ */
```
Comment (hawkw) [inline k8s/src/main/scala/io/buoyant/k8s/resources.scala]: same as above, unrelated formatting change. move this to a separate PR?

File: namerd/iface/mesh/src/main/scala/io/buoyant/namerd/iface/mesh/ResolverService.scala [hunk 3]
```diff
@@ -29,7 +29,7 @@ object ResolverService {
     }
 
     private[this] def bind(pid: mesh.Path): Var[Addr] = {
-      val id = fromPath(pid)
+      val id = Paths.stripTransformerPrefix(fromPath(pid))
```
Comment (hawkw) [inline namerd/iface/mesh/src/main/scala/io/buoyant/namerd/iface/mesh/ResolverService.scala]: lgtm 👍

File: namerd/iface/mesh/src/test/scala/io/buoyant/namerd/iface/mesh/ResolverServiceTest.scala [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      inetAf = Some(AddressFamily.INET4),
+      address = Some(Buf.ByteArray.Owned(Array[Byte](1, 2, 3, 4))),
+      port = Some(7777),
+      meta = Some(Endpoint.Meta(nodeName = None))
+    )
+    val rsp = await(resolver.getReplicas(req))
+    assert(rsp.result.get == Replicas.OneofResult.Bound(Replicas.Bound(Seq(endpoint))))
+  }
+
+  test("streamReplicas") {
```
Comment (hawkw) [inline namerd/iface/mesh/src/test/scala/io/buoyant/namerd/iface/mesh/ResolverServiceTest.scala]: TIOLI but it would be nice if these test names were more descriptive than just the name of the method being tested...

File: k8s/src/main/scala/io/buoyant/k8s/resources.scala (hunk 2 above)
Comment (deebo91) [reply k8s/src/main/scala/io/buoyant/k8s/resources.scala]: This was probably my fault and happened when I added the CustomResourceDefinition changes, sorry 😨

File: finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4StreamTransport.scala (hunk 1 above)
Comment (adleong) [reply finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4StreamTransport.scala]: scalariform
